### PR TITLE
add pull before push

### PR DESCRIPTION
**Pull request by yuzs2**
add a "git pull -r" before git push to ensure the push is done on the latest repo.

### CODE REVIEW

**Comment by yuzs2** [comment]
We added a "git pull" before "git push" in driver/git.go, this is to ensure every push happens when the local repo is synced with the remote repo. Doing this we could avoid possible bugs.

All tests passed locally:
`
all tests passed!
Removing intermediate container f1b9b08b7dde
 ---> 02c00f6ac274
Successfully built 02c00f6ac274
Successfully tagged semver-resource:latest
`

File: driver/git.go [hunk 1]
```diff
@@ -369,6 +369,15 @@ func (driver *GitDriver) writeVersion(newVersion semver.Version) (bool, error) {
 		return false, err
 	}
 
+	gitPull := exec.Command("git", "pull", "-r")
```
Comment (evanchaoli) [inline driver/git.go:372]: I think we should only run `git pull --rebase` when `git push` hit "reject" error.

**Comment by yuzs2** [comment]
all checks passed, the sign-off is not missing

**Comment by evanchaoli** [comment]
After the second look at the code, I think this PR is not needed. It already has a retry logic:

https://github.com/concourse/semver-resource/blob/d498338b21e68134fc8744d6025f8dc8a0cf326b/driver/git.go#L58

This PR is trying to resolve #124 , but I just realize the bug should be at:

https://github.com/concourse/semver-resource/blob/d498338b21e68134fc8744d6025f8dc8a0cf326b/driver/git.go#L75

As `err` is local inside the loop, it cannot be returned.

**Comment by evanchaoli** [comment]
@yuzs2 Could you please just close this PR?

**Comment by yuzs2** [comment]
> 

Yes, I'm closing it since it's no need. Thanks for the review.

**Pull request closed without merging**